inference-engine/tests/functional/plugin/shared/include/behavior/config.hpp

```diff
@@ -18,52 +18,174 @@
+    InferenceEngine::CNNNetwork cnnNet(function);
+    // Load config
+    std::map<std::string, std::string> config = {{CONFIG_KEY(EXCLUSIVE_ASYNC_REQUESTS), CONFIG_VALUE(NO)}};
+    config.insert(configuration.begin(), configuration.end());
+    if (targetDevice.find(CommonTestUtils::DEVICE_MULTI) == std::string::npos &&
+    targetDevice.find(CommonTestUtils::DEVICE_HETERO) == std::string::npos) {
+        ASSERT_NO_THROW(ie->SetConfig(config, targetDevice));
+    }
+    // Load CNNNetwork to target plugins
+    auto execNet = ie->LoadNetwork(cnnNet, targetDevice, config);
+    execNet.CreateInferRequest();
+
+    if ((targetDevice == CommonTestUtils::DEVICE_FPGA) || (targetDevice == CommonTestUtils::DEVICE_MYRIAD) ||
+    (targetDevice == CommonTestUtils::DEVICE_KEEMBAY)) {
+        ASSERT_EQ(1u, InferenceEngine::ExecutorManager::getInstance()->getExecutorsNumber());
+    } else if (targetDevice == CommonTestUtils::DEVICE_MULTI) {
+    } else {
+        ASSERT_EQ(0u, InferenceEngine::ExecutorManager::getInstance()->getExecutorsNumber());
+    }
+    config.clear();
+}
+
+TEST_P(CorrectConfigAPITests, reusableCPUStreamsExecutor) {
+    // Skip test according to plugin specific disabledTestPatterns() (if any)
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    ASSERT_EQ(0u, InferenceEngine::ExecutorManager::getInstance()->getExecutorsNumber());
+    ASSERT_EQ(0u, InferenceEngine::ExecutorManager::getInstance()->getIdleCPUStreamsExecutorsNumber());
+
+    // Create CNNNetwork from ngrpah::Function
+    InferenceEngine::CNNNetwork cnnNet(function);
+    {
+        // Load config
+        std::map<std::string, std::string> config = {{CONFIG_KEY(EXCLUSIVE_ASYNC_REQUESTS), CONFIG_VALUE(NO)}};
+        config.insert(configuration.begin(), configuration.end());
+        if (targetDevice.find(CommonTestUtils::DEVICE_MULTI) == std::string::npos &&
+        targetDevice.find(CommonTestUtils::DEVICE_HETERO) == std::string::npos) {
+            ASSERT_NO_THROW(ie->SetConfig(config, targetDevice));
+        }
+        // Load CNNNetwork to target plugins
+        auto execNet = ie->LoadNetwork(cnnNet, targetDevice, config);
+        execNet.CreateInferRequest();
+
+        if ((targetDevice == CommonTestUtils::DEVICE_FPGA) || (targetDevice == CommonTestUtils::DEVICE_MYRIAD) ||
+        (targetDevice == CommonTestUtils::DEVICE_KEEMBAY)) {
+            ASSERT_EQ(1u, InferenceEngine::ExecutorManager::getInstance()->getExecutorsNumber());
+            ASSERT_EQ(0u, InferenceEngine::ExecutorManager::getInstance()->getIdleCPUStreamsExecutorsNumber());
+        } else if (targetDevice == CommonTestUtils::DEVICE_MULTI) {
+        } else {
+            ASSERT_EQ(0u, InferenceEngine::ExecutorManager::getInstance()->getExecutorsNumber());
+            ASSERT_GE(2u, InferenceEngine::ExecutorManager::getInstance()->getIdleCPUStreamsExecutorsNumber());
+        }
+        config.clear();
+    }
+    if (targetDevice == CommonTestUtils::DEVICE_CPU) {
```

Comment:
What is it??? Looks like additional code

Reply:
@iefode  If you are talking about the IF construction, then this is from the old infrastructure.